### PR TITLE
Minimal test harness for iOS

### PR DESCRIPTION
**Pull request by dnfield**
To run:

```
autoninja -C out/ios_debug_sim_unopt FlutterTests
open -a Simulator.app
xcrun simctl install booted out/ios_debug_sim_unopt/FlutterTests.app
```

Then use either Console.app or `xcrun simctl spawn booted log stream`, launch the app, and look at the log output.

It would probably be nicer to bootstrap this from XCTest instead.

Known issues:

- springboard reruns the test when it exits currently, so you get all your logs twice.
- not clear how to capture the exit code, and we're not obviously printing it anywhere that I can see.

### CODE REVIEW

File: build/darwin/ios_app.gni [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+template("ios_app") {
```
Comment (gw280) [inline build/darwin/ios_app.gni]: can we add an `if (is_ios)` here and assert if we're not ios and trying to use this template?

File: shell/platform/darwin/ios/FlutterTestsInfo.plist [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  <key>CFBundleExecutable</key>
+  <string>FlutterTests</string>
+  <key>CFBundleIdentifier</key>
+  <string>io.flutter.FlutterTests</string>
+  <key>CFBundleInfoDictionaryVersion</key>
+  <string>6.0</string>
+  <key>CFBundleName</key>
+  <string>FlutterTests</string>
+  <key>CFBundleVersion</key>
+  <string>1.0</string>
```
Comment (gw280) [inline shell/platform/darwin/ios/FlutterTestsInfo.plist]: do we need to worry about the fact that these version strings are hardcoded?

File: testing/run_all_unittests.cc [hunk 3]
```diff
@@ -2,9 +2,22 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include "flutter/fml/build_config.h"
+#include "flutter/fml/logging.h"
```
Comment (chinmaygarde) [inline testing/run_all_unittests.cc]: Nit: Unnecessary include of logging.h

File: shell/platform/darwin/ios/flutter_unittests.mm [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+#include "flutter/testing/testing.h"
+
+namespace flutter_tests {
+
+TEST(SmokeTest, Success) {
+  EXPECT_EQ(1, 1);
+}
+
+TEST(SmokeTest, Fail) {
```
Comment (chinmaygarde) [inline shell/platform/darwin/ios/flutter_unittests.mm]: This requires a different harness.

File: shell/platform/darwin/ios/BUILD.gn [hunk 5]
```diff
@@ -249,6 +251,30 @@ shared_library("copy_and_verify_framework_headers") {
...
+    "flutter_unittests.mm",
+  ]
+
+  libs = [
+    "Foundation.framework",
+  ]
+
+  deps = [
+    ":flutter_tests_fixtures",
+    "$flutter_root/runtime:libdart",
```
Comment (chinmaygarde) [inline shell/platform/darwin/ios/BUILD.gn]: We can stop using `flutter_root` now. If you still want to use it though, please use it consistently in rules (the next dep does not use the variable here).

File: build/darwin/ios_app.gni (hunk 1 above)
Comment (chinmaygarde) [reply build/darwin/ios_app.gni]: Good call. You can also just say `assert(is_ios)`

File: testing/run_all_unittests.cc [hunk 6]
```diff
@@ -2,9 +2,22 @@
...
 #include "gtest/gtest.h"
 
+#ifdef OS_IOS
+#include <asl.h>
+#endif  // OS_IOS
+
 int main(int argc, char** argv) {
+#ifdef OS_IOS
+  asl_log_descriptor(NULL, NULL, ASL_LEVEL_NOTICE, STDOUT_FILENO,
```
Comment (chinmaygarde) [inline testing/run_all_unittests.cc]: In this harness, both can be ASL_LEVEL_ERR. I don't think notices are shown by default.

File: build/darwin/ios_app.gni (hunk 1 above)
Comment (dnfield) [reply build/darwin/ios_app.gni]: I think `assert(is_ios)` is fine, but we can't do what I think @gw280 is asking - we need to use this template from iOS.

File: shell/platform/darwin/ios/FlutterTestsInfo.plist (hunk 2 above)
Comment (dnfield) [reply shell/platform/darwin/ios/FlutterTestsInfo.plist]: For the case of witing a unit testing app, no.  For a real app, we'd probably want better handling of the plist in general.

File: testing/run_all_unittests.cc (hunk 6 above)
Comment (dnfield) [reply testing/run_all_unittests.cc]: In a local test they were, but I'll just make them both err.

File: shell/platform/darwin/ios/flutter_unittests.mm (hunk 4 above)
Comment (dnfield) [reply shell/platform/darwin/ios/flutter_unittests.mm]: Right now this test isn't run and nothing is tested.  I wanted to see if we could somehow get the exit code from this or what log scraping might make sense for CI.

File: testing/run_all_unittests.cc (hunk 3 above)
Comment (dnfield) [reply testing/run_all_unittests.cc]: Done

**Pull request merged**